src/main/io/frsky_osd.c

```diff
@@ -0,0 +1,927 @@
+
+#define FRSKY_OSD_CHAR_DATA_BYTES 54
+#define FRSKY_OSD_CHAR_METADATA_BYTES 10
+#define FRSKY_OSD_CHAR_TOTAL_BYTES (FRSKY_OSD_CHAR_DATA_BYTES + FRSKY_OSD_CHAR_METADATA_BYTES)
+
+#define FRSKY_OSD_SEND_BUFFER_SIZE 192
+#define FRSKY_OSD_RECV_BUFFER_SIZE 128
+
+#define FRSKY_OSD_CMD_RESPONSE_ERROR 0
+
+#define FRSKY_OSD_INFO_INTERVAL_MS 1000
+
+#define FRSKY_OSD_TRACE(...)
+#define FRSKY_OSD_DEBUG(...)
+#define FRSKY_OSD_ERROR(...)
+#define FRSKY_OSD_ASSERT(x)
+
+typedef enum
+{
+    OSD_CMD_RESPONSE_ERROR = 0,
+
+    OSD_CMD_INFO = 1,
+    OSD_CMD_READ_FONT = 2,
+    OSD_CMD_WRITE_FONT = 3,
+    OSD_CMD_GET_CAMERA = 4,
+    OSD_CMD_SET_CAMERA = 5,
+    OSD_CMD_GET_ACTIVE_CAMERA = 6,
+    OSD_CMD_GET_OSD_ENABLED = 7,
+    OSD_CMD_SET_OSD_ENABLED = 8,
+
+    OSD_CMD_TRANSACTION_BEGIN = 16,
+    OSD_CMD_TRANSACTION_COMMIT = 17,
+    OSD_CMD_TRANSACTION_BEGIN_PROFILED = 18,
+    OSD_CMD_TRANSACTION_BEGIN_RESET_DRAWING = 19,
+
+    OSD_CMD_DRAWING_SET_STROKE_COLOR = 22,
+    OSD_CMD_DRAWING_SET_FILL_COLOR = 23,
+    OSD_CMD_DRAWING_SET_STROKE_AND_FILL_COLOR = 24,
+    OSD_CMD_DRAWING_SET_COLOR_INVERSION = 25,
+    OSD_CMD_DRAWING_SET_PIXEL = 26,
+    OSD_CMD_DRAWING_SET_PIXEL_TO_STROKE_COLOR = 27,
+    OSD_CMD_DRAWING_SET_PIXEL_TO_FILL_COLOR = 28,
+    OSD_CMD_DRAWING_SET_STROKE_WIDTH = 29,
+    OSD_CMD_DRAWING_SET_LINE_OUTLINE_TYPE = 30,
+    OSD_CMD_DRAWING_SET_LINE_OUTLINE_COLOR = 31,
+
+    OSD_CMD_DRAWING_CLIP_TO_RECT = 40,
+    OSD_CMD_DRAWING_CLEAR_SCREEN = 41,
+    OSD_CMD_DRAWING_CLEAR_RECT = 42,
+    OSD_CMD_DRAWING_RESET = 43,
+    OSD_CMD_DRAWING_DRAW_BITMAP = 44,
+    OSD_CMD_DRAWING_DRAW_BITMAP_MASK = 45,
+    OSD_CMD_DRAWING_DRAW_CHAR = 46,
+    OSD_CMD_DRAWING_DRAW_CHAR_MASK = 47,
+    OSD_CMD_DRAWING_DRAW_STRING = 48,
+    OSD_CMD_DRAWING_DRAW_STRING_MASK = 49,
+    OSD_CMD_DRAWING_MOVE_TO_POINT = 50,
+    OSD_CMD_DRAWING_STROKE_LINE_TO_POINT = 51,
+    OSD_CMD_DRAWING_STROKE_TRIANGLE = 52,
+    OSD_CMD_DRAWING_FILL_TRIANGLE = 53,
+    OSD_CMD_DRAWING_FILL_STROKE_TRIANGLE = 54,
+    OSD_CMD_DRAWING_STROKE_RECT = 55,
+    OSD_CMD_DRAWING_FILL_RECT = 56,
+    OSD_CMD_DRAWING_FILL_STROKE_RECT = 57,
+    OSD_CMD_DRAWING_STROKE_ELLIPSE_IN_RECT = 58,
+    OSD_CMD_DRAWING_FILL_ELLIPSE_IN_RECT = 59,
+    OSD_CMD_DRAWING_FILL_STROKE_ELLIPSE_IN_RECT = 60,
+
+    OSD_CMD_CTM_RESET = 80,
+    OSD_CMD_CTM_SET = 81,
+    OSD_CMD_CTM_TRANSLATE = 82,
+    OSD_CMD_CTM_SCALE = 83,
+    OSD_CMD_CTM_ROTATE = 84,
+    OSD_CMD_CTM_ROTATE_ABOUT = 85,
+    OSD_CMD_CTM_SHEAR = 86,
+    OSD_CMD_CTM_SHEAR_ABOUT = 87,
+    OSD_CMD_CTM_MULTIPLY = 88,
+
+    OSD_CMD_CONTEXT_PUSH = 100,
+    OSD_CMD_CONTEXT_POP = 101,
+
+    // MAX7456 emulation commands
+    OSD_CMD_DRAW_GRID_CHR = 110,
+    OSD_CMD_DRAW_GRID_STR = 111,
+} osdCommand_e;
+
+typedef enum {
+    RECV_STATE_NONE,
+    RECV_STATE_SYNC,
+    RECV_STATE_LENGTH,
+    RECV_STATE_DATA,
+    RECV_STATE_CHECKSUM,
+    RECV_STATE_DONE,
+} frskyOSDRecvState_e;
+
+typedef struct frskyOSDInfoResponse_s {
```

Comment:
Commonly used form for camelCase would be `frskyOsdInfoResponse_s` (to avoid awkward capitalisations in situations like `oSDInit`).

Reply:
I'll do a global `s/frskyOSD/frskyOsd/g`